tests/integration/mirrorpeersecret_controller_test.go

```diff
@@ -0,0 +1,534 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	multiclusterv1alpha1 "github.com/red-hat-storage/odf-multicluster-orchestrator/api/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/common"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	rtscheme "k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	pkgscheme "sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+type MirrorPeerPlusPlus multiclusterv1alpha1.MirrorPeer
+
+func (mppp MirrorPeerPlusPlus) GetMirrorPeer() multiclusterv1alpha1.MirrorPeer {
+	return multiclusterv1alpha1.MirrorPeer(mppp)
+}
+
+func (mppp MirrorPeerPlusPlus) GetManagedClusters() []clusterv1.ManagedCluster {
+	var managedClusters []clusterv1.ManagedCluster
+	for _, eachItem := range mppp.Spec.Items {
+		managedCluster := clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: eachItem.ClusterName,
+			},
+			Spec: clusterv1.ManagedClusterSpec{},
+		}
+		managedClusters = append(managedClusters, managedCluster)
+	}
+	return managedClusters
+}
+
+func (mppp MirrorPeerPlusPlus) GetClusterNamespaces() []corev1.Namespace {
+	uniqueNamespaceMap := map[string]struct{}{}
+	for _, eachPeer := range mppp.Spec.Items {
+		uniqueNamespaceMap[eachPeer.ClusterName] = struct{}{}
+	}
+	var namespaceArr []corev1.Namespace
+	for namespace := range uniqueNamespaceMap {
+		namespace := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		namespaceArr = append(namespaceArr, namespace)
+	}
+	return namespaceArr
+}
+
+func (mppp MirrorPeerPlusPlus) GetDummySourceSecrets(prefixName string) []corev1.Secret {
+	var destSecrets []corev1.Secret
+	for i, eachPR := range mppp.Spec.Items {
+		secretNN := types.NamespacedName{
+			Name:      fmt.Sprintf("%s_%d", prefixName, i),
+			Namespace: eachPR.ClusterName,
+		}
+		storageClusterNN := types.NamespacedName{
+			Name:      eachPR.StorageClusterRef.Name,
+			Namespace: eachPR.StorageClusterRef.Namespace,
+		}
+		destSecret := common.CreateSourceSecret(secretNN, storageClusterNN, []byte("MySecretData1234"))
```

Comment:
Shouldn't variable name be `sourceSecret`?

Reply:
yes... changed it